### PR TITLE
Totals use fewer tables when possible

### PR DESCRIPTION
**Pull request by LindsayYoung**
This make the call a bit faster by automatically reducing the amount of tables if the call does not need them all. I also reduced the amount of fields that show up automatically.

### CODE REVIEW

File: webservices/db.py [hunk 1]
```diff
@@ -26,3 +26,18 @@ def htsql_conn():
         htsql_conn_string = sqla_conn_string().replace('postgresql', 'pgsql')
         g._htsql_conn = HTSQL(htsql_conn_string)
     return g._htsql_conn
+
+def as_dicts(data):
```
Comment (cmc333333) [inline webservices/db.py]: If you're moving this here (seems like it makes more sense), remember to remove the version in `webservices/resources.py`

File: webservices/totals/models.py [hunk 2]
```diff
@@ -48,13 +40,48 @@ def query_text(self, show_fields):
...
+                    ?cmte_id={'%s'}"%(
+                        "','".join(ids)
+                )
+            data = htsql_conn().produce(pre_qry)
+            types = []
+            # looking for the codes in the htsql string
+            for d in str(data):
+                if d.isalpha() == True:
+                    if d not in types:
+                        types.append(d)
```
Comment (cmc333333) [inline webservices/totals/models.py]: Don't know if you'd like this one-liner better:

```python
types = set(d for d in str(data) if d.isalpha())
```

**Comment by cmc333333** [comment]
I don't understand the underlying data well enough to validate, but it looks like this divides what type of query it makes based on the request, which no doubt removes a lot of unneeded work. :+1:

File: webservices/totals/models.py (hunk 2 above)
Comment (LindsayYoung) [inline webservices/totals/models.py]: Good point. I will do that.

File: webservices/db.py (hunk 1 above)
Comment (LindsayYoung) [inline webservices/db.py]: Good catch, I am doing that now!

**Comment by LindsayYoung** [comment]
@cmc333333 thanks for the suggestions. They should all be incorporated now.

I also fixed #337 while I was in there and added a test. The API will return a sane format when no search results are found.

**Comment by LindsayYoung** [comment]
just merged this in to the older pr about totals

**Pull request closed without merging**